packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt

```diff
@@ -315,70 +311,44 @@ class SyncSessionTests {
+    // SyncSessions available inside the error handler are disconnected from the underlying
     // Realm instance and some APIs could have a difficult time understanding semantics. For now, we
     // just disallow calling these APIs from these instances.
     @Test
-    @Ignore // TODO remove ignore when the this is fixed https://github.com/realm/realm-kotlin/issues/867
     fun syncSessionFromErrorHandlerCannotUploadAndDownloadChanges() = runBlocking {
-        val channel = Channel<SyncSession>(1)
-        var wrongSchemaRealm: Realm? = null
-        val job = async {
-
-            // Create server side Realm with one schema
-            var config = SyncConfiguration.Builder(
-                user,
-                partitionValue,
-                schema = setOf(ParentPk::class, ChildPk::class)
-            ).build()
-            val realm = Realm.open(config)
-            realm.syncSession.uploadAllLocalChanges()
-            realm.close()
-
-            // Create same Realm with another schema, which will cause a Client Reset.
-            config = SyncConfiguration.Builder(
-                user,
-                partitionValue,
-                schema = setOf(ParentPk::class, io.realm.kotlin.entities.sync.bogus.ChildPk::class)
-            ).name("new_realm.realm")
-                .syncClientResetStrategy(object : DiscardUnsyncedChangesStrategy {
-                    @Suppress("TooGenericExceptionThrown")
-                    override fun onBeforeReset(realm: TypedRealm) {
-                        // TODO This approach doesn't work on native until these callbacks return a
-                        //  boolean - the exceptions make the thread crash on Kotlin Native
-                        throw Exception("Land on onError")
-                    }
+        // Open Realm with a user that has no read nor write permissions
+        // See 'canWritePartition' in TestAppInitializer.kt.
+        val (email, password) = "test_nowrite_noread_${TestHelper.randomEmail()}" to "password1234"
+        val user = runBlocking {
+            app.createUserAndLogIn(email, password)
+        }
+        val channel = Channel<SyncSession>(1).freeze()
+        val config = SyncConfiguration.Builder(
+            schema = setOf(ParentPk::class, ChildPk::class),
+            user = user,
+            partitionValue = partitionValue
+        ).errorHandler { session, _ ->
+            channel.trySend(session)
+        }.build()
 
-                    override fun onAfterReset(before: TypedRealm, after: MutableRealm) {
-                        // Writing "fail("whatever") makes no sense here because Core will redirect
-                        // the execution to onError below
+        runBlocking {
+            val deferred = async { Realm.open(config) }
+            val session = channel.receive()
+            try {
+                assertFailsWithMessage<IllegalStateException>("Operation is not allowed inside a `SyncSession.ErrorHandler`.") {
```

Comment:
This error message does not tell that the `Uploading and downloading changes is not allowed` are we sure that we are testing it right?